File: Easy/JavaScript/482. License Key Formatting.js
var licenseKeyFormatting = function(s, k) {
    let license = [];

    for(let i = s.length - 1; 0 <= i; i--){
        if(s.charAt(i) != '-'){
            if(license.length % (k + 1) == k)
                license.push('-');
            license.push(s.charAt(i));
        }
    }

    return license.reverse().join("").toUpperCase();
};

File: Easy/Java/482. License Key Formatting.java
class Solution {
    public String licenseKeyFormatting(String s, int k) {
        StringBuilder license = new StringBuilder();

        for(int i = s.length() - 1; 0 <= i; i--){
            if(s.charAt(i) != '-'){
                if(license.length() % (k + 1) == k)
                    license.append('-');
                license.append(s.charAt(i));
            }
        }

        return license.reverse().toString().toUpperCase();
    }
}
